schema_editor/app/scripts/views/recordtype/related-aggregates-directive.js

```diff
@@ -0,0 +1,178 @@
+(function() {
+    'use strict';
+
+    /* ngInject */
+    function RTRelatedAggregatesController(
+        $state, $scope, $stateParams, Schemas, RecordTypes, RecordSchemas, RecordCosts,
+        Notifications
+    ) {
+        var ctl = this;
+        ctl.$onInit = onInit;
+
+        function onInit() {
+            // functions
+            ctl.submitForm = submitForm;
+            ctl.populatePropertyKeys = populatePropertyKeys;
+            ctl.populateEnumFields = populateEnumFields;
+            ctl.reset = reset;
+
+            // data
+            ctl.selectedContentType = {};
+            ctl.selectedPropertyKey = {};
+            ctl.definitions = {};
+
+            RecordTypes.get({
```

Comment:
It might be possible to clean this promise chaining up a bit so that it's not quite so deeply nested if you return promises from each stage. Take a look here, this might help with refactoring: http://solutionoptimist.com/2013/12/27/javascript-promise-chains-2/